### PR TITLE
Improve checks in release script

### PR DESCRIPTION
**Pull request by liry**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: release.sh [hunk 1]
```diff
@@ -34,46 +34,58 @@ done
...
+if [[ -z "${VERSION}" ]]; then
    echo "No version given!"
    usage
    exit 1
 fi
 
-if [ -z "${GPG_PASS}" ]
-then
+if ! [[ ${VERSION} =~ ^[0-9\.]+$ ]]; then
+    echo "Version has to match regex ^[0-9\.]+$"
```
Comment (jimirocks) [inline release.sh]: This will prevent RC from releasing

Comment (jimirocks) [reply release.sh]: and also it matches things like `.`, `000`, `5.` etc...

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: release.sh (hunk 1 above)
Comment (liry) [reply release.sh]: fixed

**Comment by jimirocks** [comment]
please also port it RC...

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/55/ : SUCCESS in 18s

**Pull request merged**